### ISSUE TITLE
How to run on windows 10

### ISSUE THREAD

**Issue by senthilsam** [mathjax/MathJax-demos-node#7]
I am not able to run the scripts on windows 10.

**Comment by dpvc** [comment]
You have not given enough information to be able to help you out.  What have you tried?  What error messages are you getting?

The files are set up as unix command-line scripts, but on Windows, you may have to call them using the node executable. Try

    node -r esm [file] [arguments]

e.g.,

    node -r esm components/tex2mml 'x^2+y^2=z^2'

**Comment by senthilsam** [comment]
Thanks for the response.
When i run the command with help message everything is fine,

e.g.

**command:**
`node -r esm speech\mml2svg --help`

**output:**
```
mml2svg [options] "math" > file.html

Options:
  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]
  --em         em-size in pixels                                   [default: 16]
  --ex         ex-size in pixels                                    [default: 8]
  --width      width of container in pixels                      [default: 1280]
  --speech     level of speech: deep, shallow, none         [default: "shallow"]
  --css        output the required CSS rather than the SVG itself      [boolean]
  --fontCache  whether to use a local font cache or not[boolean] [default: true]
```

But when i pass the mathml as argument I am getting the below error.

**command:**
`node -r esm speech\mml2svg "<math altimg='si2.svg'><msub><mi>I</mi><mrow><mi>z</mi><mo>−</mo><msub><mi>b</mi><mn>1</mn></msub></mrow></msub><mo>=</mo><munder><mo>∫</mo><mi>A</mi></munder><msup><mi>y</mi><mn>2</mn></msup><mi mathvariant='italic'>dA</mi><mo linebreak='goodbreak'>=</mo><msubsup><mo>∫</mo><mrow><mo>−</mo><mi>b</mi><mo>/</mo><mn>2</mn></mrow><mrow><mi>b</mi><mo>/</mo><mn>2</mn></mrow></msubsup><mi>dz</mi><mrow><mfenced open='[' close=']'><mrow><msubsup><mo>∫</mo><mrow><mo>−</mo><mi>h</mi><mo>/</mo><mn>2</mn></mrow><mrow><mo>−</mo><mi>d</mi><mo>/</mo><mn>2</mn></mrow></msubsup><msup><mi>y</mi><mn>2</mn></msup><mi>d</mi><mi mathvariant='normal'>y</mi><mo linebreak='badbreak'>+</mo><msubsup><mo>∫</mo><mrow><mi>d</mi><mo>/</mo><mn>2</mn></mrow><mrow><mi>h</mi><mo>/</mo><mn>2</mn></mrow></msubsup><msup><mi>y</mi><mn>2</mn></msup><mi mathvariant='italic'>dy</mi></mrow></mfenced></mrow><mo linebreak='goodbreak'>=</mo><mfrac><mi>b</mi><mn>12</mn></mfrac><mrow><mfenced open='(' close=')'><mrow><msup><mi>h</mi><mn>3</mn></msup><mo>‐</mo><msup><mi>d</mi><mn>3</mn></msup></mrow></mfenced></mrow></math>"`

**output:**
`C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\util\Retries.js:1
Error: MathJax retry
    at Object.retryAfter (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\util\Retries.js:24:15)
    at class_1.EnrichedMathItemMixin.class_1.enrich (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\a11y\semantic-enrich.js:75:38)
    at Object.renderMath (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\core\MathDocument.js:106:65)
    at RenderList.renderConvert (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\core\MathDocument.js:158:35)
    at class_1.AbstractMathItem.convert (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\core\MathItem.js:46:32)
    at class_2.AbstractMathDocument.convert (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\core\MathDocument.js:297:15)
    at Object.<anonymous> (C:\Users\205647\Downloads\MathJax-demos-node-master\speech\mml2svg:94:19)
    at Generator.next (<anonymous>)`

I don't know what is happening here.

**Comment by dpvc** [comment]
It works for me.  Did you do the `npm install` that is indicated in the [installation instructions](https://github.com/mathjax/MathJax-demos-node#installation)?

**Comment by senthilsam** [comment]
Yep. I did ;(

**Comment by dpvc** [comment]
OK, I've checked it out, and it is a problem with MathJax's handling of the Windows paths with `\` instead of unix-style `/` directory separators.

Try replacing the contents of `node_modules/mathjax-full/js/util/asyncLoad./node.js` with the following:

```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var mathjax_js_1 = require("../../mathjax.js");
var path = require("path");
var root = path.dirname(path.dirname(__dirname));
if (!mathjax_js_1.mathjax.asyncLoad && typeof require !== 'undefined') {
    mathjax_js_1.mathjax.asyncLoad = function (name) {
        return require(name.charAt(0) === '.' ? path.resolve(root, name) : name);
    };
}
//# sourceMappingURL=node.js.map
```

and see if that helps.

**Comment by senthilsam** [comment]
@dpvc  You saved my life. This perfectly works for the file  `speech\mml2svg`

But when run the file `component\mml2svg` Again i am facing issues.

**command:**
`>node -r esm component\mml2svg "<math altimg='si2.svg'><msub><mi>I</mi><mrow><mi>z</mi><mo>−</mo><msub><mi>b</mi><mn>1</mn></msub></mrow></msub><mo>=</mo><munder><mo>∫</mo><mi>A</mi></munder><msup><mi>y</mi><mn>2</mn></msup><mi mathvariant='italic'>dA</mi><mo linebreak='goodbreak'>=</mo><msubsup><mo>∫</mo><mrow><mo>−</mo><mi>b</mi><mo>/</mo><mn>2</mn></mrow><mrow><mi>b</mi><mo>/</mo><mn>2</mn></mrow></msubsup><mi>dz</mi><mrow><mfenced open='[' close=']'><mrow><msubsup><mo>∫</mo><mrow><mo>−</mo><mi>h</mi><mo>/</mo><mn>2</mn></mrow><mrow><mo>−</mo><mi>d</mi><mo>/</mo><mn>2</mn></mrow></msubsup><msup><mi>y</mi><mn>2</mn></msup><mi>d</mi><mi mathvariant='normal'>y</mi><mo linebreak='badbreak'>+</mo><msubsup><mo>∫</mo><mrow><mi>d</mi><mo>/</mo><mn>2</mn></mrow><mrow><mi>h</mi><mo>/</mo><mn>2</mn></mrow></msubsup><msup><mi>y</mi><mn>2</mn></msup><mi mathvariant='italic'>dy</mi></mrow></mfenced></mrow><mo linebreak='goodbreak'>=</mo><mfrac><mi>b</mi><mn>12</mn></mfrac><mrow><mfenced open='(' close=')'><mrow><msup><mi>h</mi><mn>3</mn></msup><mo>‐</mo><msup><mi>d</mi><mn>3</mn></msup></mrow></mfenced></mrow></math>"`

**Output:**
`MathJax(adaptors/liteDOM): Cannot find module 'mathjax-full/es5/C:\\Users\\205647\\Downloads\\MathJax-demos-node-master\\node_modules\\mathjax-full\\components\\src/adaptors/liteDOM/liteDOM.js'
Require stack:
- C:\Users\205647\Downloads\MathJax-demos-node-master\component\mml2svg
`
I am very new js itself. so can't figure out what is the issue. BTW i have also node the changes in `ts/util/asyncLoad/node.ts` as mentioned in the above link which you have commented.

**Comment by dpvc** [comment]
OK, I'll look into it.  In the meantime, 

    node -r esm component\mml2svg --dist "<math ... "

works for me.  Also the corresponding `preload` and `direct` versions work, so you could use one of those for now.

**Comment by dpvc** [comment]
OK, found the problem.  If you want to patch it locally, edit the file `node_modules\mathjax-full\js\components\package.js` and look for `Package.resolvePath`, which should be line 74.  Three lines after that you will see
```
        if (!file.match(/^(?:[a-z]+:\/)?\/|\[/)) {
```

with you should change to

```
        if (!file.match(/^(?:[a-z]+:\/)?\/|\[|[a-z]:\\/i)) {
```

That should take care of it.

**Comment by dpvc** [comment]
Note that the change is to add `|[a-z]:\\` just before the final `/`, and to add `i` after the `/`.  Both these are needed.

**Comment by senthilsam** [comment]
Thanks a lot. You save all my hair pulling moments:(.
Perfectly works now!!!

**Comment by senthilsam** [comment]
I am getting below error only if i convert this mathml equation.

**command:**
`>node -r esm speech\mml2svg "<math altimg='si3.svg'><mtable columnalign='center' displaystyle='true'><mtr><mtd><mi mathvariant='normal'>d</mi><mo>'</mo><mi>où</mi><mspace width='0.25em'/><mi>x</mi></mtd><mtd><mo linebreak='newline'>=</mo></mtd><mtd><mfrac><mrow><mn>2</mn><mo>×</mo><mn>36</mn></mrow><mn>15</mn></mfrac></mtd><mtd/></mtr><mtr><mtd><mspace width='1.2em'/><mi>x</mi></mtd><mtd><mo linebreak='newline'>=</mo></mtd><mtd><mn>4</mn><mo>,</mo><mn>8</mn><mspace width='0.25em'/></mtd><mtd><mi mathvariant='italic'>soit</mi><mspace width='0.25em'/><mn mathvariant='italic'>4</mn><mo>,</mo><mn mathvariant='italic'>8</mn><mspace width='0.25em'/><mi mathvariant='italic'>kg</mi><mspace width='0.25em'/><mi mathvariant='italic'>de</mi><mi mathvariant='italic'>peinture</mi></mtd></mtr></mtable></math>"`

**output:**
`C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:1
Error: Unknown node type "mtd/"
    at MathMLCompile.error (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:220:15)
    at MathMLCompile.makeNode (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:69:49)
    at MathMLCompile.addChildren (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:129:57)
    at MathMLCompile.makeNode (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:76:14)
    at MathMLCompile.addChildren (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:129:57)
    at MathMLCompile.makeNode (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:76:14)
    at MathMLCompile.addChildren (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:129:57)
    at MathMLCompile.makeNode (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:76:14)
    at MathMLCompile.compile (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml\MathMLCompile.js:40:24)
    at MathML.compile (C:\Users\205647\Downloads\MathJax-demos-node-master\node_modules\mathjax-full\js\input\mathml.js:81:82)`

**Comment by dpvc** [comment]
This is issue mathjax/MathJax-src#328, which is that self-closing tags aren't handled properly by the internal liteDOM module that is used when you run MAthJax in node.  This is fixed in PR mathjax/MathJax-src#370, which will be in the next release.  For now, you need to use `<mtd></mtd>` instead of `<mtd/>`.

**Comment by dpvc** [comment]
Also, you should not reuse an old issue tracker for a new issue.  Please start a fresh issue to report a new problem that you are having.
